sympy/core/numbers.py

```diff
@@ -1042,6 +1042,10 @@ def __new__(cls, num, dps=None, prec=None, precision=None):
                 # it's a hexadecimal (coming from a pickled object)
                 # assume that it is in standard form
                 num = list(num)
+                # long uses a shim for int on Python 3, so we need to
+                # remove any trailing 'L' on the num[1] string
```

Comment:
I'm not sure anyone will understand this comment later...

How about:
````
# If we're loading an object pickled in Python 2 into Python 3, we may need to
# strip a tailing 'L' because of a shim for int on Python 3, see issue #13470.
````

Other than this bikeshedding ;-), this LGTM.